crates/cli/src/benchmark.rs

```diff
@@ -166,25 +185,48 @@ impl InProcessBenchmarkCommand {
         let bytes = fs::read(&self.wasmfile).context("Attempting to read Wasm bytes")?;
         log::debug!("Wasm benchmark size: {} bytes", bytes.len());
 
-        let mut measurements = Vec::with_capacity(self.iterations);
+        let arch = this_arch();
+        let engine = self.engine.to_string();
+        let wasm = self.wasmfile.display().to_string();
+        let mut measurements = Measurements::new(arch, &engine, &wasm);
+
+        let mut measure = self.measure.build();
+
         for _ in 0..self.iterations {
-            measurements.push(benchmark(&mut bench_api, &bytes, self.measure)?);
+            benchmark(&mut bench_api, &bytes, &mut measure, &mut measurements)?;
+            measurements.next_iteration();
```

Comment:
Here's what I was trying to say earlier but wasn't too clear about: why not avoid passing the mutable `Measurements` all the way down into the sightglass-recorder crate? The overhead of returning a `Vec` is not important, but this would simplify the recorder logic. Instead, the "flatten to CSV rows" logic could live in the sightglass-data crate and this code would look roughly like:

```
let ms = benchmark(&mut bench_api, &bytes, &mut measure?);
measurements.insert_multiple(ms);
```